torchvision/transforms/_functional_pil.py

```diff
@@ -310,6 +311,19 @@ def rotate(
     if not _is_pil_image(img):
         raise TypeError(f"img should be PIL Image. Got {type(img)}")
 
+    angle = angle - math.floor(angle / 360) * 360  # shift angle to [0, 360) range
+
+    # fast path: transpose without affine transform
+    if expand or center is None:
+        if angle == 0:
+            return img
+        elif angle == 90:
+            return img.transpose(Image.ROTATE_90)
+        elif angle == 180:
+            return img.transpose(Image.ROTATE_180)
+        elif angle == 270:
+            return img.transpose(Image.ROTATE_270)
```

Comment:
Pillow already does that in image.rotate implementation:
https://github.com/python-pillow/Pillow/blob/8f63748e50378424628155994efd7e0739a4d1d1/src/PIL/Image.py#L2287-L2296